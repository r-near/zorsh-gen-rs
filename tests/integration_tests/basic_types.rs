use anyhow::Result;

#[test]
fn test_primitive_types() -> Result<()> {
    let input = r#"
        #[derive(BorshSerialize)]
        struct Numbers {
            u8_val: u8,
            u16_val: u16,
            u32_val: u32,
            u64_val: u64,
            i8_val: i8,
            i16_val: i16,
            i32_val: i32,
            i64_val: i64,
            f32_val: f32,
            f64_val: f64,
        }
    "#;

    let output = zorsh_gen_rs::convert_str(input)?;
    insta::assert_snapshot!(output);
    Ok(())
}

#[test]
fn test_string_types() -> Result<()> {
    let input = r#"
        #[derive(BorshSerialize)]
        struct User {
            name: String,
            email: String,
        }
    "#;

    let output = zorsh_gen_rs::convert_str(input)?;
    insta::assert_snapshot!(output);
    Ok(())
}

#[test]
fn test_unit_structs() -> Result<()> {
    let input = r#"
        #[derive(BorshSerialize)]
        struct Empty;

        #[derive(BorshSerialize)]
        struct EmptyBraces {}
    "#;

    let output = zorsh_gen_rs::convert_str(input)?;
    insta::assert_snapshot!(output);
    Ok(())
}

#[test]
fn test_basic_enums() -> Result<()> {
    let input = r#"
        #[derive(BorshSerialize)]
        enum Status {
            Active,
            Inactive,
            Pending,
        }
    "#;

    let output = zorsh_gen_rs::convert_str(input)?;
    insta::assert_snapshot!(output);
    Ok(())
}
