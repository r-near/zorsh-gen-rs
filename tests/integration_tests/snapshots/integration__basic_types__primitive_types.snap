---
source: tests/integration_tests/basic_types.rs
expression: output
snapshot_kind: text
---
import { b } from '@zorsh/zorsh';

export const NumbersSchema = b.struct({
    u8_val: b.u8(),
    u16_val: b.u16(),
    u32_val: b.u32(),
    u64_val: b.u64(),
    i8_val: b.i8(),
    i16_val: b.i16(),
    i32_val: b.i32(),
    i64_val: b.i64(),
    f32_val: b.f32(),
    f64_val: b.f64()
});
export type Numbers = b.infer<typeof NumbersSchema>;
